### PR TITLE
Increase test coverage of test_utils.py on Python 3

### PR DESCRIPTION
**Pull request by rodrigc**
See:
https://twistedmatrix.com/trac/ticket/8582

### CODE REVIEW

File: twisted/test/test_iutils.py [hunk 1]
```diff
@@ -121,8 +120,8 @@ def test_outputAndValue(self):
 
         def gotOutputAndValue(out_err_code):
             out, err, code = out_err_code
-            self.assertEqual(out, "hello world!\n")
-            self.assertEqual(err, "goodbye world!" + os.linesep)
+            self.assertEqual(out, b"hello world!\n")
+            self.assertEqual(err, b"goodbye world!" + os.linesep.encode())
```
Comment (Lukasa) [inline twisted/test/test_iutils.py]: Can we use `.encode('ascii')` please? I don't want us to trip up over weird encodings.

File: twisted/test/test_iutils.py [hunk 2]
```diff
@@ -226,7 +227,7 @@ def _defaultPathTest(self, utilFunc, check):
         os.chmod(dir, 0)
 
         d = utilFunc(self.exe, ['-u', scriptFile])
-        d.addCallback(check, dir)
+        d.addCallback(check, dir.encode())
```
Comment (Lukasa) [inline twisted/test/test_iutils.py]: I'm nervous about these two naked `encode` calls as well. Is there a sensible encoding we can choose here? Maybe `sys.getfilesystemencoding()`?

File: twisted/test/test_iutils.py (hunk 1 above)
Comment (rodrigc) [inline twisted/test/test_iutils.py]: OK

File: twisted/test/test_iutils.py (hunk 2 above)
Comment (rodrigc) [inline twisted/test/test_iutils.py]: OK

**Pull request merged**